spec/models/solidus_volume_pricing/pricer_spec.rb

```diff
@@ -0,0 +1,280 @@
+        let(:quantity) { 6 }
+
+        before do
+          variant.volume_price_models << create(:volume_price_model)
+          variant.volume_price_models.first.volume_prices.create!(amount: 2, discount_type: 'dollar', range: '(5+)')
+        end
+
+        it 'uses the volume price from model' do
+          is_expected.to eq('$8.00')
+        end
+      end
+    end
+
+    context 'discount_type = percent' do
+      before do
+        variant.volume_prices.create!(amount: 0.25, discount_type: 'percent', range: '(10+)')
+      end
+
+      context 'when quantity does not match the range' do
+        it_behaves_like 'having the variant price'
+      end
+
+      context 'when quantity matches the range' do
+        let(:quantity) { 10 }
+
+        it 'uses the volume price' do
+          is_expected.to eq('$7.50')
+        end
+
+        context 'when volume price has a role' do
+          before do
+            variant.volume_prices.first.update(role_id: role.id)
+          end
+
+          context 'when no user is given' do
+            it_behaves_like 'having the variant price'
+          end
+
+          context 'when a user is given' do
+            let(:pricing_options) do
+              SolidusVolumePricing::PricingOptions.new(
+                quantity: quantity,
+                user: user
+              )
+            end
+
+            context 'whose role matches' do
+              before { user.spree_roles << role }
+
+              it 'uses the volume price' do
+                is_expected.to eq('$7.50')
+              end
+            end
+
+            context 'whose role does not match' do
+              before { user.spree_roles << other_role }
+
+              it_behaves_like 'having the variant price'
+            end
+          end
+        end
+      end
+
+      context 'when it does match from a volume price model' do
+        let(:quantity) { 6 }
+
+        it 'uses the volume price from model' do
+          variant.volume_price_models << create(:volume_price_model)
+          variant.volume_price_models.first.volume_prices.create!(amount: 0.75, discount_type: 'percent', range: '(5+)')
+          is_expected.to eq('$2.50')
+        end
+      end
+    end
+
+    context 'discount_type is unknown' do
+      before do
+        variant.volume_prices.create(amount: 7, discount_type: 'foo', range: '(10+)')
+      end
+
+      it_behaves_like 'having the variant price'
+    end
+
+    context 'when use_master_variant_volume_pricing' do
+      let(:master) { variant.product.master }
+
+      before do
+        @before = Spree::Config.use_master_variant_volume_pricing
+        master.volume_prices.create!(amount: 1.5, discount_type: 'price', range: '(1+)')
+        variant.volume_prices.create!(amount: 3.5, discount_type: 'price', range: '(1+)')
+      end
+
+      context 'is enabled' do
+        before do
+          Spree::Config.use_master_variant_volume_pricing = true
```

Comment:
_always_ use around here. When mutating global data in tests we have to be extremely careful to avoid any kind of leak, and we want to make it obvious we're guarding against it.

```ruby
let(:use_master_variant_volume_pricing) { true }
around do |example|
  current_pricing = Spree::Config.use_master_variant_volume_pricing
  Spree::Config.use_master_variant_volume_pricing = use_master_variant_volume_pricing
  example.run
  Spree::Config.use_master_variant_volume_pricing = current_pricing
end
```

You then override the let in your contexts as required and put this at the top level.